astroquery/jplspec/core.py

```diff
@@ -0,0 +1,246 @@
+
+        return response
+
+    def _parse_result(self, response, verbose=False):
+        """
+        Parse a response into an `~astropy.table.Table`
+
+        The catalog data files are composed of 80-character card images, with
+        one card image per spectral line.  The format of each card image is:
+        FREQ, ERR, LGINT, DR,  ELO, GUP, TAG, QNFMT,  QN',  QN"
+        (F13.4,F8.4, F8.4,  I2,F10.4,  I3,  I7,    I4,  6I2,  6I2)
+
+        FREQ:  Frequency of the line in MHz.
+        ERR:   Estimated or experimental error of FREQ in MHz.
+        LGINT: Base 10 logarithm of the integrated intensity in units of nm^2 MHz at
+            300 K.
+
+        DR:    Degrees of freedom in the rotational partition function (0 for atoms,
+            2 for linear molecules, and 3 for nonlinear molecules).
+
+        ELO:   Lower state energy in cm^{-1} relative to the ground state.
+        GUP:   Upper state degeneracy.
+        TAG:   Species tag or molecular identifier.
+            A negative value flags that the line frequency has
+            been measured in the laboratory.  The absolute value of TAG is then the
+            species tag and ERR is the reported experimental error.  The three most
+            significant digits of the species tag are coded as the mass number of
+            the species.
+
+        QNFMT: Identifies the format of the quantum numbers
+        QN':   Quantum numbers for the upper state.
+        QN":   Quantum numbers for the lower state.
+        """
+
+        result = ascii.read(response.text,
+                            header_start=None,
+                            data_start=0,  # start at 0 since regex was applied
+                            # Warning for a result with more than 1000 lines:
+                            # THIS form is currently limited to 1000 lines.
+                            comment=r'THIS|^\s{12,14}\d{4,6}.*',
+                            names=('FREQ', 'ERR', 'LGINT', 'DR', 'ELO', 'GUP',
+                                   'TAG', 'QNFMT', 'QN\'', 'QN"'),
+                            col_starts=(0, 13, 21, 29, 31, 41, 44, 51, 55, 67),
+                            format='fixed_width')
+
+        if len(result) > self.maxlines:
+            warnings.warn("This form is currently limited to {0} lines.\n\
+                          Please limit your search.".format(self.maxlines))
+
+        result['FREQ'].unit = u.MHz
+        result['ERR'].unit = u.MHz
+        result['LGINT'].unit = u.nm**2 * u.MHz
+        result['ELO'].unit = u.cm**(-1)
+
+        return result
+
+    def get_species_table(self, catfile='catdir.cat'):
+        """
+        A directory of the catalog is found in a file called 'catdir.cat.'
+        Each element of this directory is an 80-character record with the
+        following format:
+
+        | TAG,  NAME, NLINE,  QLOG,  VER
+        | (I6,X, A13, I6, 7F7.4,  I2)
+
+        Parameters
+        -----------
+        catfile : default 'catdir.cat'
```

Comment:
str, name of the file, rather that a file object (I mean it's obvious as the default is a string, but can be made more consistent).